### ISSUE TITLE
"Live" link broken when editing a page in live + draft status 

### ISSUE THREAD

**Issue by kaedroho** [wagtail/wagtail#2820]
When editing a page in "live + draft" status where the slug is different to the live revision. The "live" link at the top right is broken because it is trying to direct the user to the new slug rather than the current one.

We should make this link go use the live slug, rather than the one in the revision.

**Comment by jjanssen** [comment]
Related PR #3302, which eventually makes it easier to solve.

**Comment by lb-** [comment]
I have confirmed this issue and will submit a pull request.

When exploring the parent page the status column has the button 'Live + Draft' with the correct (live slug) link. However, when editing the page itself the same button 'Live + Draft' will use the latest draft slug in the link.

It appears that the edit page view is intentionally getting the latest page revision as a page, not the 'live' version of the page.

`    page = get_object_or_404(Page, id=page_id).get_latest_revision_as_page()`
https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailadmin/views/pages.py#L297

This is then passed to the include template [page_status_tag.html](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailadmin/templates/wagtailadmin/shared/page_status_tag.html).

When this template grabs the URL it will be whatever is stored in the latest revision, not what is available on the actual live Page.

I will add the following to pages.py (inside the edit view).

```
    # Page status needs to present the version of the page containing the correct live URL
    if page.has_unpublished_changes:
        page_for_status = latest_revision.page
    else:
        page_for_status = page
```

Then pass `page_for_status` to the edit template, using it instead of `page` when using the include `page_status_tag.html`.

**Comment by gasman** [comment]
Fixed in #3988
